templates/survey.html

```diff
@@ -0,0 +1,300 @@
+                                            ]
+                                        }
+                                    ]
+                                },
+                                {
+                                    "rowTitle": "Code and Architecture",
+                                    "rowItems": [
+                                        {
+                                            "valueList": [
+                                            {
+                                                "text": "<div id='project_details'></div>"
+                                            }
+                                        ],
+                                        "id": "project-details",
+                                            "iconType": "chevron", 
+                                            "actions": [
+                                                {   
+                                                    "text": "Change",
+                                                    "ariaLabel": "View answers for Household accommodation",
+                                                    "url": url_for('architecture_pre_survey')
+                                                }
+                                            ]
+                                        }
+                                    ]
+                                },
+                                {
+                                    "rowTitle": "Technology Details",
+                                    "rowItems": [
+                                        {
+                                            "valueList": [
+                                            {
+                                                "text": "<div id='developed_details'></div>"
+                                            }
+                                        ],
+                                        "id": "developed-details",
+                                            "iconType": "chevron",
+                                            "actions": [
+                                                {
+                                                    "text": "Change",
+                                                    "ariaLabel": "View answers for Mary Smith",
+                                                    "url": url_for('tech_pre_survey')
+                                                }
+                                            ]
+                                        }
+                                    ]
+                                },
+                            ]
+                        }
+                    ]
+                }
+            ]
+        }) }}
+    {{ onsButton({
+            "id": "submit-btn",
+            "text": '<div id="submit-btn-text">Continue to Submission</div>',
+            "classes": 'ons-u-mb-m ons-u-mt-l',
+            "url": url_for('validate_details')
+        }) }}
+    <script>
```

Comment:
I think this section is a prime candidate for refactor. Can we open a JIRA ticket to take another look at this?  I suspect we can pull the scripts out of the html, introduce helper functions and make the code more readable.  Once we have done that ticket it will probably give clues to how the other html files could be refactored too.

Reply:
I pulled out the scripts for survey.html and added some helper functions, but I think it may require a bit more discussion